ffi/src/lib.rs

```diff
@@ -1,18 +1,48 @@
 // Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
 // See the file LICENSE.md for licensing terms.
 
+use std::collections::HashMap;
 use std::ffi::{CStr, CString, OsStr, c_char};
 use std::fmt::{self, Display, Formatter};
+use std::ops::Deref;
 use std::os::unix::ffi::OsStrExt as _;
 use std::path::Path;
+use std::sync::atomic::{AtomicU32, Ordering};
+use std::sync::{Arc, RwLock};
 
-use firewood::db::{BatchOp as DbBatchOp, Db, DbConfig, DbViewSync as _};
+use firewood::db::{BatchOp as DbBatchOp, Db, DbConfig, DbViewSync as _, Proposal};
 use firewood::manager::{CacheReadStrategy, RevisionManagerConfig};
 
 mod metrics_setup;
 
 use metrics::counter;
 
+type ProposalId = u32;
+
+pub struct DatabaseHandle<'p> {
+    proposals: RwLock<HashMap<ProposalId, Arc<Proposal<'p>>>>,
+    db: Db,
+    next_id: AtomicU32,
```

Comment:
@alarso16 suggested that maybe this should be a global to avoid possible errors of committing against the wrong database.